File: api/src/services/user/models/AuthenticateUser.ts
import { IsString, IsEmail } from 'class-validator';
import { IsFacebookAccessToken } from '../../../validators';

class AuthenticateUser {
  @IsString()
  @IsFacebookAccessToken()
  accessToken: string;

  @IsString()
  firstName: string;

  @IsString()
  lastName: string;

  @IsEmail()
  email: string;
}

export default AuthenticateUser;

File: api-go/models/authenticateUser.go
package models

type AuthenticateUser struct {
	AccessToken string `form:"accessToken" json:"accessToken" binding:"required"`
	FirstName   string `form:"firstName" json:"firstName" binding:"required"`
	LastName    string `form:"lastName" json:"lastName" binding:"required"`
	Email       string `form:"email" json:"email" binding:"required"`
}
